client/tests/webdriver/specs/createNewLocation.spec.js

```diff
@@ -0,0 +1,54 @@
+import { expect } from "chai"
+import CreateNewLocation from "../pages/location/createNewLocation.page"
+import {
+  BAD_LAT_LNG_VAL,
+  LOCATION_COORDS,
+  LOCATION_NAME
+} from "./locationUtils"
+
+describe("When creating a new Location", () => {
+  it("Should not create a location without name input", () => {
+    CreateNewLocation.open()
+    CreateNewLocation.createButton.click()
+    CreateNewLocation.nameRequiredError.waitForExist()
+    CreateNewLocation.nameRequiredError.waitForDisplayed()
+  })
+
+  it("Should not accept invalid latitude-longitude inputs", () => {
+    CreateNewLocation.nameField.setValue(LOCATION_NAME)
+    CreateNewLocation.latField.setValue(BAD_LAT_LNG_VAL)
+    CreateNewLocation.lngField.setValue(BAD_LAT_LNG_VAL)
+    // trigger onblur effect
+    browser.keys(["Tab"])
+
+    CreateNewLocation.latLngErrorsDisplayed()
+  })
+
+  it("Should have a location with correct MGRS in popover", () => {
+    CreateNewLocation.latField.setValue(
+      "\uE003".repeat(BAD_LAT_LNG_VAL.length) + LOCATION_COORDS.lat
+    )
+    CreateNewLocation.lngField.setValue(
+      "\uE003".repeat(BAD_LAT_LNG_VAL.length) + LOCATION_COORDS.lng
+    )
+    CreateNewLocation.allFormatsPopover.click()
+    CreateNewLocation.allFormatsPopoverLatLng.waitForExist()
+    CreateNewLocation.allFormatsPopoverMGRS.waitForExist()
+
+    expect(CreateNewLocation.allFormatsPopoverLatLng.getText()).match(
+      new RegExp(LOCATION_COORDS.lat, "g")
+    )
+    expect(CreateNewLocation.allFormatsPopoverLatLng.getText()).match(
+      new RegExp(LOCATION_COORDS.lng, "g")
+    )
+    expect(CreateNewLocation.allFormatsPopoverMGRS.getText()).match(
+      new RegExp(LOCATION_COORDS.mgrs, "g")
+    )
+  })
```

Comment:
Instead of having expect.match, can't we use an expect.toEqual?
Otherwise, if the value in the popup would be something like LOCATION_COORDS.lat+"fff", it would still match it, right?
Instead of using CreateNewLocation.allFormatsPopoverLatLng you could use something which picks up the span value for each coordinate.

Reply:
There was no easy way to target the fields, so i didn't want to depend on child, sibling relations of the html elements. That was why i chose to pick a parent node to match for regex. I could use selectors like ( :first-child + ~ etc) but i thought it wouldn't be robust for any future change. I could use something like `getLatSpan(spanText) { return ...$("span[text=spanText]")`

Reply:
I think it's ok how you did it now with :first-child etc.